docs/man/rpm-macros.7.scd

```diff
@@ -0,0 +1,760 @@
+RPM-MACROS(7)
+
+# NAME
+
+rpm-macros - RPM macro syntax
+
+# SYNOPSIS
+## Defining
+*%*_NAME_[*(*[_OPTIONS_]*)*] _BODY_
+
+## Expanding
+*%*_NAME_
+
+*%*_NAME_ [_OPTIONS_] [_ARGUMENTS_]
+
+*%{*_NAME_*}*
+
+*%{*_NAME_ [_OPTIONS_] _ARGUMENTS_*}*
+
+*%{*_NAME_*:*_ARGUMENT_*}*
+
+*%{?*_NAME_*}*
+
+*%{?*_NAME_*:*_VALUE-IF-DEFINED_*}*
+
+*%{!?*_NAME_*:*_VALUE-IF-NOT-DEFINED_*}*
+
+*%(*_SHELL-COMMAND_*)*
+
+*%[*_EXPRESSION_*]*
+
+*%[*_EXPRESSION_ *?* _VALUE-IF-TRUE_ *:* _VALUE-IF-FALSE_*]*
+
+*%{lua:*_LUA-CODE_*}*
+
+# DESCRIPTION
+
+RPM has a powerful built-in macro processor. The primary uses of macros
+are configuration and other utility functions for RPM itself,
+and as a packaging aid in spec files.
+
+In addition to simple text substitution, the macro processor supports
+the following facilities:
+- function-like parametric macros with options and arguments processing
+  and locally scoped automatic and user-defined macros
+- various built-in primitives for string processing and OS interaction
+- shell expansion
+- expression expansion
+- embedded Lua expansion
+
+The macro definition syntax is:
+
+	_NAME_[([_OPTIONS_])] _BODY_
```

Comment:
I wonder if this hairy definition could be simplified by leaving out the square brackets immediately around `OPTIONS`? As in:

```
_NAME_[(_OPTIONS_)] _BODY_
```

I suppose they're currently there because you can legally define a macro as `foo()` (i.e. without any options). But is that actually something that's useful by itself?

Reply:
... yep, separating the two makes a whole lot more sense. Pushed a fixup to address this throughout, having the parametric macro definition in the PARAMETRIC MACROS section makes much more sense.

Reply:
Oh right, good catch (as for separating). And indeed, "empty" options are rather common, as one would notice just by looking at some real macros :sweat_smile: